disperser/dataapi/subgraph_client.go

```diff
@@ -174,23 +174,38 @@ func (sc *subgraphClient) QueryNumBatchesByOperatorsInThePastBlockTimestamp(ctx
 		intervalEventsPool   = workerpool.New(maxWorkerPoolSize)
 		currentTs            = uint64(time.Now().Unix())
 	)
+	type timeInterval struct {
+		start, end uint64
+	}
+	// Caching the number of batches in a time interval so we don't need to query
+	// subgraph repeatedly. In usual case, most operators will have no opt-in/opt-out
+	// events in recent time window, so all of them will just query the same time
+	// interval.
+	numBatchesCache := make(map[timeInterval]int)
 	for _, ie := range intervalEvents {
```

Comment:
are `ie`'s guaranteed to have no overlaps?

Reply:
Yep